compiler/src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -1295,9 +1293,20 @@ object Parsers {
         case _: (ForYield | ForDo) => in.token == FOR
         case _ => false
 
+      def matchesWithUpdated(stat: Tree): Boolean = {
```

Comment:
"Updated" makes me think that the same statement might might be updated multiple times but as far as I can tell we only need to set the end marker once, so I would call this matchesAndSet

Reply:
I changed it to `matchedAndSetEnd`